From dc1feaa5b3827ad6120f8b1544b5c4fbfea800a0 Mon Sep 17 00:00:00 2001
From: Nic Costa <nic.costa@gmail.com>
Date: Fri, 17 Apr 2020 15:46:08 -0500
Subject: [PATCH] HACK: wait for the rpc client to call PT Register

Async RPC calls won't receive replies from edge-core if the
client hasn't first completed pt registration.
---
 tls/tpm.go | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/tls/tpm.go b/tls/tpm.go
index 5768bb8..cb2aa14 100644
--- a/tls/tpm.go
+++ b/tls/tpm.go
@@ -185,7 +185,10 @@ func newJSONRPCClient(socket string, path string) *rpc.Client {
 		return nil
 	}
 
-	return rpc.Dial(socket, path, onConn)
+	client := rpc.Dial(socket, path, onConn)
+	fmt.Printf("HACK: sleeping for 5 seconds while the client registers")
+	time.Sleep(5 * time.Second)
+	return client
 }
 
 func (pk *ecdsaPrivateKey) Public() crypto.PublicKey {
-- 
2.17.1

